src/main/java/com/uber/jenkins/phabricator/PhabricatorNotifier.java

```diff
@@ -231,6 +236,23 @@ public final boolean perform(final AbstractBuild<?, ?> build, final Launcher lau
         return true;
     }
 
+    protected UberallsClient getUberallsClient(Logger logger, String gitUrl, String branch) {
+        if (uberallsClient != null) {
+            return uberallsClient;
+        }
+        return new UberallsClient(
+                getDescriptor().getUberallsURL(),
+                logger,
+                gitUrl,
+                branch
+        );
+    }
+
+    // Just for testing
+    protected void setUberallsClient(UberallsClient client) {
```

Comment:
I'm not sure this is the preferred pattern. Can't you just use mockito or powermock to mock the call to `getUberallsClient` and then eliminate the class member?